File: src/Services/BudgetCast.Expenses/src/BudgetCast.Expenses.Queries/Expenses/GetExpenseById/ExpenseDetailsVm.cs
namespace BudgetCast.Expenses.Queries.Expenses.GetExpenseById
{
    public class ExpenseDetailsVm
    {
        public long Id { get; set; }

        public string AddedBy { get; set; }

        public DateTime AddedAt { get; set; }

        public DateTime CreatedOn { get; set; }

        public string Description { get; set; }

        public long CampaignId { get; set; }

        public string[] Tags { get; set; }

        public IReadOnlyList<ExpenseItemDetailsVm> ExpenseItems { get; set; }

        public ExpenseDetailsVm()
        {
            AddedBy = default!;
            Description = default!;
            Tags = default!;
            ExpenseItems = default!;
        }
    }
}

File: src/Web/WebSPA/BudgetCast.Spa/BudgetCast.Spa/Client/src/app/modules/receipt-management/pages/models/expense-details-vm.ts
import { ExpenseItemDetailsVm } from './expense-item-details-vm';

export interface ExpenseDetailsVm {
    id: number;
    campaignId: number;
    addedBy: string;
    createdOn: string;
    addedAt: Date;
    description: string;
    tags: string[];
    expenseItems: ExpenseItemDetailsVm[];
}